src/ui/api/repository.go

```diff
@@ -416,7 +416,7 @@ func (ra *RepositoryAPI) GetTopRepos() {
 		ra.CustomAbort(http.StatusBadRequest, "invalid count")
 	}
 
-	repos, err := dao.GetTopRepos(count)
+	repos, err := dao.GetTopRepos(ra.GetSession("userId"), count)
```

Comment:
Use function GetUserIDForRequest in src/common/api/base.go to get ID of user. It supports authenticating user against both basic auth and session.
BTW, open the PR to branch dev please.